core/services/headtracker/head_tracker.go

```diff
@@ -132,7 +130,7 @@ func (ht *HeadTracker) Start() error {
 		}
 
 		ht.wgDone.Add(3)
-		go ht.headListener.ListenForNewHeads(ht.handleNewHead)
+		go ht.headListener.ListenForNewHeads(ht.handleNewHead, ht.wgDone.Done)
```

Comment:
This tracker/listener stuff wasn't a race, but I thought it looked suspicious and was able to simplify it.